kibana/resources.py

```diff
@@ -127,6 +132,64 @@ def find_elastic(cls, **params):
         params = cls._add_internal_filter(True, params)
         return cls.find(**params)
 
+    @classmethod
+    def bulk_action(cls, action: definitions.RuleBulkActions, rule_ids: Optional[List[str]] = None,
+                    query: Optional[str] = None, dry_run: Optional[bool] = False,
+                    edit: Optional[definitions.RuleBulkEditAction] = None,
```

Comment:
The question here:

between this and `bulk_edit` and `bulk_duplicate`

The way this is written would require them to import those objects from `kibana.definitions`, set them, and pass them  in. Alternatively, we could expose the parameters directly, and then just do the validation (load the objects) from _within_ these methods.

So, essentially, what is a better experience

Reply:
It really depends on if you're talking about better experience for maintainers or users. I recommend requiring imported objects for maintainability. We'll have more flexibility long term, even if there's bit more boilerplate code. 

If you want to make the experience better for users/callers, then exposing directly is the way to go. It seems simpler, and the validation like you mentioned is centralized.

Reply:
yea ... I know .... and I agree .... was just trying to convince myself not to rework it 😅 . I'll update

Reply:
lol so funny. I SAID to Eric. I think @brokensound77 already knows the answer to this. lol